ui/app/components/file-to-array-buffer.js

```diff
@@ -15,54 +17,45 @@ import filesize from 'filesize';
  * ```js
  *   <FileToArrayBuffer @onChange={{action (mut file)}} />
  * ```
- * @param onChange=null {Function} - The function to call when the file read is complete. This function
- * receives the file as a JS ArrayBuffer
- * @param [label=null {String}] - Text to use as the label for the file input
- * @param [fileHelpText=null {String} - Text to use as help under the file input
+ * @param {function} onChange - The function to call when the file read is complete. Receives the file as a JS ArrayBuffer
+ * @param {string} [label] - Text to use as the label for the file input
+ * @param {string} [fileHelpText] - Text to use as help under the file input
  *
  */
-export default Component.extend({
-  classNames: ['box', 'is-fullwidth', 'is-marginless', 'is-shadowless'],
-  onChange: () => {},
-  label: null,
-  fileHelpText: null,
-
-  file: null,
-  filename: null,
-  fileSize: null,
-  fileLastModified: null,
+export default class FileToArrayBufferComponent extends Component {
+  @tracked filename = null;
+  @tracked fileSize = null;
+  @tracked fileLastModified = null;
 
   readFile(file) {
     const reader = new FileReader();
-    // raft-snapshot-restore test was failing on CI trying to send action on destroyed object
-    // ensure that the component has not been torn down prior to sending onChange action
-    if (!this.isDestroyed && !this.isDestroying) {
-      reader.onload = () => this.send('onChange', reader.result, file);
-    }
```

Comment:
It looks to me like this was where the test flakiness was coming from. I recall the test failing frequently after the upgrade and wrapping the entire block in the destroyed check must have improved things but really the check should be inside the `onload` callback which would fire sometime in the future.